### PR TITLE
Fix crash open 4

### PR DESCRIPTION
**Pull request by demonoidv**
Fix the crash that happens when we have an open four.
This crash was happening because we didn't check the end of the game the right way and we didn't calculate the last heuristic (for the node that can end the game).

This PR aims to fix that behavior.

WARNING: PR #48 must be merged first in order to allow me to deal with the conflict on my branch directly.

### CODE REVIEW

File: src/algorithm.rs [hunk 1]
```diff
@@ -176,27 +183,27 @@ impl Algorithm
             .map(|b| {
                 let mut player_captures = parent_player_captures;
                 let mut enemy_captures = parent_enemy_captures;
-                let (player, enemy) =
+                let (player, enemy, is_players_move) =
```
Comment (Ragarnoy) [inline src/algorithm.rs]: For now OK but this doesn't make the code easy to read tbh

Comment (Ragarnoy) [reply src/algorithm.rs]: oops meant to make this a comment lol

**Review by Ragarnoy — APPROVED** [review]
Merge after #48

**Review by arlaine4 — COMMENTED** [review]
stonks

**Review by arlaine4 — APPROVED** [review]
stonks

**Pull request merged**